### PR TITLE
Fix: remove the setTransitionTime from cluster status controller.

### PR DESCRIPTION
**Pull request by xigang**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

There is no need to call setTransitionTime before calling the SetStatusCondition method. From the code logic, call setTransitionTime is not required.

https://github.com/karmada-io/karmada/blob/master/pkg/controllers/status/cluster_status_controller.go#L180 and https://github.com/karmada-io/karmada/blob/master/pkg/controllers/status/cluster_status_controller.go#L205

**Which issue(s) this PR fixes**:
Fixes #https://github.com/karmada-io/karmada/issues/3313

**Special notes for your reviewer**:

@XiShanYongYe-Chang 

**Does this PR introduce a user-facing change?**:
```
NONE
```

### CODE REVIEW

**Review by Poor12 — COMMENTED** [review]
Could you delete `setTransitionTime` together? Seems this function is non-used.

**Comment by xigang** [comment]
> Could you delete `setTransitionTime` together? Seems this function is non-used.

`setTransitionTime` has been deleted

**Review by RainbowMango — COMMENTED** [review]
Can you have a test on it to see if the condition transition time is set as expected?

It should be ok, but one more test would be better.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks~
/lgtm

**Comment by XiShanYongYe-Chang** [comment]
Have you make a test on your local site?

**Comment by xigang** [comment]
> Have you made a test on your local site?

I add some unit test cases

> Have you made a test on your local site?
I'll add some unit test cases a little later.

**Comment by XiShanYongYe-Chang** [comment]
Hi, @xigang thanks for your work. I think we don't need to add ut to the method in the third-party library, but judging from your new ut context, what @RainbowMango says is satisfied.

Can you help remove the newly added ut, and we can continue to approve?

File: pkg/util/condition_test.go [hunk 1]
```diff
@@ -56,3 +58,102 @@ func TestIsConditionsEqual(t *testing.T) {
...
+					LastTransitionTime: metav1.Unix(int64(1679496434), int64(0)),
+				},
+			},
+			expected: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			meta.SetStatusCondition(tt.ouputConditions, tt.inputCondition)
```
Comment (RainbowMango) [inline pkg/util/condition_test.go:153]: We don't need this unit test, as this function is maintained by Kubernetes and there is already a unit test at:
https://github.com/kubernetes/kubernetes/blob/d2be69ac11346d2a0fab8c3c168c4255db99c56f/staging/src/k8s.io/apimachinery/pkg/api/meta/conditions_test.go#L27

**Review by RainbowMango — CHANGES_REQUESTED** [review]
Sorry, I didn't make it clear.

> Can you have a test on it to see if the condition transition time is set as expected?

I mean we can test it manually would be fine.  No worries, I can help do it.
But, please remove the unit test.

**Comment by xigang** [comment]
@XiShanYongYe-Chang @RainbowMango The newly added unit tests have been removed. I only have the local environment, can you help verify it in your test environment?

**Review by RainbowMango — APPROVED** [review]
It works. Thanks.
/lgtm
/approve

Ready condition:
```yaml
  - lastTransitionTime: "2023-03-23T02:01:33Z"
    message: cluster is healthy and ready to accept workloads
    reason: ClusterReady
    status: "True"
    type: Ready
```

Not Ready condition:
```yaml
  - lastTransitionTime: "2023-03-23T02:06:53Z" ## We can see the transitionTime is updated as expected.
    message: cluster is not reachable
    reason: ClusterNotReachable
    status: "False"
    type: Ready
```

**Comment by xigang** [comment]
> It works. Thanks. /lgtm /approve
> 
> Ready condition:
> 
> ```yaml
>   - lastTransitionTime: "2023-03-23T02:01:33Z"
>     message: cluster is healthy and ready to accept workloads
>     reason: ClusterReady
>     status: "True"
>     type: Ready
> ```
> 
> Not Ready condition:
> 
> ```yaml
>   - lastTransitionTime: "2023-03-23T02:06:53Z" ## We can see the transitionTime is updated as expected.
>     message: cluster is not reachable
>     reason: ClusterNotReachable
>     status: "False"
>     type: Ready
> ```

@RainbowMango  Thanks for helping verify.

**Pull request merged**